src/main/java/org/assertj/core/api/AbstractClassAssert.java

```diff
@@ -647,4 +647,59 @@ public SELF hasPublicMethods(String... methodNames) {
     classes.assertHasPublicMethods(info, actual, methodNames);
     return myself;
   }
+
+  /**
+   * Verifies that the actual {@code Class} has the given package name.
+   * <p>
+   * Example:
+   * <pre><code>
+   * package one.two;
+   *
+   * class MyClass {}
+   *
+   * // this assertions succeeds:
+   * assertThat(MyClass.class).hasPackage("one.two");
+   *
+   * // these assertions fail:
+   * assertThat(MyClass.class).hasPackage("one");
+   * assertThat(MyClass.class).hasPackage("");
+   * assertThat(MyClass.class).hasPackage("java.lang");</code></pre>
+   *
+   * @param packageName the package name which the class must declare
+   * @return {@code this} assertions object
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} doesn't declare the given package.
+   */
+  public SELF hasPackage(String packageName) {
+    classes.assertHasPackage(info, actual, packageName);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the given package.
```

Comment:
As we are still java 8 based and `Class#getPackageName()` comes from java 9+, I'd explicitly mention the relationship with `Class#getPackage()` behavior:

```suggestion
   * Verifies that the actual {@code Class} has the given package (as in {@link Class#getPackage()}).
   *
```